_posts/2020-07-12-amiko-hydras-proof.markdown

```diff
@@ -0,0 +1,598 @@
+where the innermost argument list uses \\\(N - 1\\\) copies of \\\(K(H)\\\).
+Let \\\(b_1 = 0, a_1 = K(E)\\\).
+If we expand the \\\(i\\\), the outermost expression uses \\\(b_0 = 0\\\) if \\\(N=1\\\) and otherwise \\\(b_0=i_{F, 0:g(F,D)}(0, K(H), K(H), \cdots, K(H), 0)\\\) with \\\(N-2\\\) copies of \\\(K(H)\\\), and \\\(a_0 = K(H)\\\).
+Recall \\\(H = S(E)\\\).
+Then \\\(a_0 = K(H) = K(S(E)) \lessapprox K(E) = a_1\\\), which we can assume because \\\(E\\\) is a label hydra and we are proving this assuming already that label hydras decrease.
+We also need to show \\\((b_0 \lessapprox^c b_1 \lor \exists x \in c\cup \\\{b_1\\\},b_0 \lessapprox x)\\\), where \\\(c\\\) is the set of parent expressions visited plus \\\(b_1\\\) itself, by finding a suitable \\\(x\\\).
+Let \\\(x\\\) be the \\\(K\\\) of the rightmost child of \\\(F\\\).
+Then \\\(b_0, x\\\) look the same until we reach the node corresponding to \\\(D\\\).
+If there are no further layers, then \\\(b_0 = 0\\\) and it is obvious we are done.
+Otherwise, recurse again, choosing the same \\\(x\\\).
+It can be inductively shown then that \\\((b_0 \lessapprox^c b_1 \lor \exists x \in c\cup \\\{b_1\\\},b_0 \lessapprox x)\\\) for all \\\(N\\\).
+\\\(a_0 \lessapprox^c a_1 \land b_0 \lessapprox^c b_1\\\) implies \\\(i_{F, 0:g(F,D)}(0, K(H), K(H), \cdots, K(H), 0) \lessapprox^c h_{F, 0:g(F,D)}(0, K(E))\\\), which then implies \\\(K(S(A)) \lessapprox K(A)\\\).
+Thus, in this case, a sub-expression does decrease.
+
+</div>
+
+## Rule 1.2
+{: #proof-rule-12}
+
+{% include collapser.markdown %}
+
+<div>
+
+Suppose a suitable \\\(E\\\) is found.
+Let \\\(E^\* \\\) be \\\(E\\\) but with label changed to \\\(\star\\\).
+Then, \\\(K(E^\* ) = (a_0, b_0, (a_1, b_1, \cdots (a_{j-1}, b_{j-1}, 0) \cdots ))\\\) for some \\\(a, b\\\).
+Due to the relative locations, \\\(K(D)\\\) must appear as a sub-expression of \\\(b_0\\\) (or possibly it is \\\(b_0\\\)).
+Let \\\(i = \min \\\{i | i\in j \land 0 \lessapprox a_i \\\} \cup \\\{ j \\\}\\\).
+Note that \\\(\forall k\in i, a_i = 0\\\).
+Since a suitable \\\(E\\\) was found, and we know \\\(D\\\) and \\\(E\\\) must both have label \\\((\star:)\\\), we can say
+\\\((a_i, b_i, (a_{i+1}, b_{i+1}, \cdots (a_{j-1}, b_{j-1}, 0) \cdots )) \lessapprox K(D')\\\).
+Let \\\(E'^\* \\\) be \\\(E'\\\) but with label changed to \\\(\star\\\).
+Let \\\(D''\\\) be \\\(S(D')\\\) but with children with label \\\((\star:)\\\) removed.
+We know \\\(K(D'') \lessapprox K(S(D'))\\\) since removing children causes a sub-expression to decrease, and \\\(K(S(D')) \lessapprox K(D')\\\) by induction.
+Then, \\\(K(D'') \lessapprox K(D')\\\).
+Observe \\\(K(E'^\* ) = (a_0, b_0, (a_1, b_1, \cdots (a_{i-1}, b_{i-1}, K(D'')) \cdots )) = (0, b_0, (0, b_1, \cdots (0, b_{i-1}, K(D'')) \cdots ))\\\).
+Observe that the transform "Replace \\\(F\\\) with \\\(E'\\\) where \\\(D\\\) is replaced by \\\(F\\\)."
```

Comment:
You may want to take out the period at the end of this line, it looks like the sentence ends.

Reply:
Sure thing